src/Query/JoinClause.php

```diff
@@ -116,15 +125,36 @@ public function using(...$columns): self
     }
 
     /**
-     * Alias for using method.
+     * Set "on" clause for join.
      *
-     * @param array ...$columns
+     * @param Closure|string $first
+     * @param mixed|null     $operator
+     * @param mixed|null     $second
+     * @param string         $boolean
```

Comment:
As i see there is no `$boolean` parameter and there is no `Closure` support for `$first` parameter.

Reply:
Also i think it is better to list only supported types for all parameters instead of a `mixed` typehint.